plugin/completion/lib_complete.py

```diff
@@ -109,12 +109,11 @@ def remove(self, view_id):
             view_id (int): view id
 
         """
-        with self.rlock:
-            if view_id not in self.TUs:
-                log.error(" no tu for view id: %s, so not removing", view_id)
-                return
-            log.debug(" removing translation unit for view id: %s", view_id)
-            del self.TUs[view_id]
+        if view_id not in self.TUs:
+            log.error(" no tu for view id: %s, so not removing", view_id)
+            return
+        log.debug(" removing translation unit for view id: %s", view_id)
+        del self.TUs[view_id]
```

Comment:
With no lock, this will risk accessing TU concurrently which is not supported by libclang and can cause crashes in the plugin process.